main.go

```diff
@@ -126,6 +130,55 @@ func main() {
     return c.JSONPretty(http.StatusOK, &responseBody, "  ")
   })
 
+  e.GET("/generate-invoice", func(c echo.Context) error {
+		responseBody := &GIResponse{}
+
+    ln := c.QueryParam("ln")
+    lnurlpUrl, _, err := ToUrl(ln)
+    if err != nil {
+      return c.JSON(http.StatusBadRequest, &responseBody)
+    }
+
+    lnurlp, lnurlpResponse, err := GetJSON(lnurlpUrl)
+    if err != nil {
+      e.Logger.Errorf("%v", err)
+    }
+
+    // if the request resulted in error return a bad request. something must be wrong with the ln address
+    if lnurlpResponse == nil {
+      return c.JSON(http.StatusBadRequest, &responseBody)
+    }
+    // if the response have no success
+    if lnurlpResponse != nil && lnurlpResponse.StatusCode > 300 {
+      return c.JSONPretty(lnurlpResponse.StatusCode, &responseBody, "  ")
+    }
+		callback := lnurlp.(map[string]interface{})["callback"]
+
+		// if the lnurlp response doesn't have a callback to generate invoice
+		if callback == nil {
+			return c.JSON(http.StatusBadRequest, &responseBody)
+		}
+
+    invoiceParams := c.QueryParam("invoiceParams")
+
+		invoice, invoiceResponse, err := GetJSON(callback.(string) + "?" + invoiceParams);
```

Comment:
does this work? is `invoiceParams` an object? 
I would think of the URL as something like this: 
```
proxyurl.com?ln=satoshi@getalby.com&invoice[amount]=1000&invoice[comment]=foo%20bar
```
and then we pass on everything that's in `invoice`

is this how this works ?

Reply:
otherwise I fear it is double encoded. and it is harder to first convert the invoice params into a string

Reply:
Done!
Now it takes in `http://localhost:3000/generate-invoice?ln=adithya@getalby.com&amount=1000&comment=foo`

And deletes the `ln` param, encodes the rest again and appends it to the callback